probe/probe.go

```diff
@@ -302,3 +334,362 @@ func probeNode(c *k8s.Client) {
+	if err != nil {
+		return false, err
+	}
+	color.Green("\nFound KubeArmor running in Kubernetes \n\n")
+	_, err = boldWhite.Printf("Daemonset : \n")
+	if err != nil {
+		color.Red(" Error while printing")
+	}
+	desired, ready, available := w.Status.DesiredNumberScheduled, w.Status.NumberReady, w.Status.NumberAvailable
+	if desired != ready && desired != available {
+		data = append(data, []string{"\t kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
+		return false, nil
+	}
+	data = append(data, []string{"\t kubearmor ", "Desired: " + strconv.Itoa(int(desired)), "Ready: " + strconv.Itoa(int(ready)), "Available: " + strconv.Itoa(int(available))})
+	renderOutputInTableWithNoBorders(data)
+	return true, nil
+}
+
+func getKubeArmorDeployments(c *k8s.Client) {
+	var data [][]string
+	_, err := boldWhite.Printf("Deployments : \n")
+	if err != nil {
+		color.Red(" Error while printing")
+	}
+
+	//relay deployment
+	relayDeployment, err := c.K8sClientset.AppsV1().Deployments("kube-system").Get(context.Background(), "kubearmor-relay", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	//not updated replicas- what we need is desired replicas
+	desired1, ready1, available1 := relayDeployment.Status.UpdatedReplicas, relayDeployment.Status.ReadyReplicas, relayDeployment.Status.AvailableReplicas
+	data = append(data, []string{"\t KubeArmor-Relay: ", "Desired: " + strconv.Itoa(int(desired1)), "Ready: " + strconv.Itoa(int(ready1)), "Available: " + strconv.Itoa(int(available1))})
+
+	//host policy manager deployment
+	hostPolicyDeployment, err := c.K8sClientset.AppsV1().Deployments("kube-system").Get(context.Background(), "kubearmor-host-policy-manager", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	//not updated replicas- what we need is desired replicas
+	desired2, ready2, available2 := hostPolicyDeployment.Status.UpdatedReplicas, hostPolicyDeployment.Status.ReadyReplicas, hostPolicyDeployment.Status.AvailableReplicas
+	data = append(data, []string{"\t KubeArmor-Host-Policy-Manager: ", "Desired: " + strconv.Itoa(int(desired2)), "Ready: " + strconv.Itoa(int(ready2)), "Available: " + strconv.Itoa(int(available2))})
+
+	//policy manager deployment
+	policyManagerDeployment, err := c.K8sClientset.AppsV1().Deployments("kube-system").Get(context.Background(), "kubearmor-policy-manager", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	//not updated replicas- what we need is desired replicas
+	desired3, ready3, available3 := policyManagerDeployment.Status.UpdatedReplicas, policyManagerDeployment.Status.ReadyReplicas, policyManagerDeployment.Status.AvailableReplicas
+	data = append(data, []string{"\t KubeArmor-Policy-Manager: ", "Desired: " + strconv.Itoa(int(desired3)), "Ready: " + strconv.Itoa(int(ready3)), "Available: " + strconv.Itoa(int(available3))})
```

Comment:
Let's not hardcode this,

What if we fetch all the the deployments which have label selector, loop through the result and append things.

Relevant kubectl query
```sh
~❯❯❯  kubectl get pods -n kube-system -l kubearmor-app
NAME                                             READY   STATUS    RESTARTS   AGE
kubearmor-annotation-manager-5b88fc49c-tlxds     2/2     Running   0          4m29s
kubearmor-host-policy-manager-5967dc9545-stlbr   2/2     Running   0          4m33s
kubearmor-nw78b                                  1/1     Running   0          4m33s
kubearmor-policy-manager-67dbf69bfd-98lb5        2/2     Running   0          4m33s
kubearmor-relay-5656cc5bf7-567wj                 1/1     Running   0          4m33s
```

Also let's not hardcode the "kube-system" I believe we take an option as a flag, so let's pass o.Namespace instead of "kube-system"